### DISCUSSION TITLE
How should I answer the doodba copier template questions to generate a development environment?

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#456] [Q&A] [answered]
How's everybody?
I ask this question because it is something recurring and the answer is unclear. I don't know about you, but I like to be able to test all the environments on my computer first, even the production one. Then I find the need for the development, production and test environments to coexist.
In several discussions, it is made clear that the development environment must be different from the test/production environment.
In the same project, if we run a development environment and then the production environment or the other way around, significant messes are generated where images have to be reconstructed and volumes purged.
Is it possible to work on the same computer with both modes (dev and test/prod)? What would be the correct procedure to achieve this without having to rebuild images and purge volumes?

We are clear about how to create a test/production environment by answering all copier questions including Traefik, email and backup questions.
But what about the questions, if we want to build a separate development environment? Do I need to answer all the same questions about Traefik, mail and backup? Is there some parameter that makes the copier understand that we only want a development environment?

How do you work to modify code in development mode and test the test/production modes before pushing to the real server?

**Comment by pedrobaeza** [comment]
I don't understand your need: development environment is the same as at code level than the prod/test if you get latest code with `invoke git-aggregate`. You don't need to double check things.

**Reply by pleirb** [reply]
I understand, but the image generated for development is different from the one for production/testing, right? And for this reason it is necessary and mandatory to create two projects (folders) with different names, one for development and another for production/testing, if we want to test everything on the same computer.
I don't really understand the command, `invoke git-aggregate`, but I have observed that it is only useful for the development environment. To obtain code in production/test mode, I understand that you have to rebuild the image with: `docker compose build`

In any case, I don't know if you understand my concern yet, the English language is still a problem for me. What I want to do is work in both available modes, right? Development and production/test, and I have observed that the questions are more for the latter environment. To work in development mode there are questions that are irrelevant, could it be? For example, the name of the database does not matter which one you enter, in development mode it will be called **devel**. And it doesn't matter what password you use for the database, in development mode it will use **odoopassword**
Although, I wonder what happens if in the copier I respond as **devel** for the database name and **odoopassword** for the password. Can both modes be executed without any conflict?

**Reply by pleirb** [reply, marked as answer]
Going back to the initial question, someone correct me, if I'm wrong, for a development project, do it in a completely separate folder and with a different name than the test/production project if you plan to use the same computer for this purpose, this It will save you a lot of headaches that were analyzed in these two posts:
#60
tecnativa/doodba#418
And regarding the questions for development: `copier copy gh:Tecnativa/doodba-copier-template ./ --trust`
You only need to answer these questions:
**odoo version
Language
OCI/Docker image (optional)
Postgres version (optional)**
All other questions can be left blank or unanswered
